Spike for faster recovery

digitalstain opened the pull request.

fickludd committed:
Remove unused QueryState.queryId and PipeMonitor

digitalstain committed:
Improve recovery performance by smartly scanning for high id

chrisvest committed:
Small cleanups.

digitalstain committed:
Adds 10ms warning to buffer allocation for bolt responses

digitalstain committed:
Remove contention on Monitor creation

digitalstain committed:
Add cached pool for bolt session workers

jimwebber commented:
@digitalstain some JavaDoc on `Monitors#newMonitor` would be helpful guidance for future API users not to fall into the same unwary trap.

digitalstain committed:
Thread naming and pooling improvements in bolt worker thread pool

chrisvest committed:
Add Apache Commons Compression dependency to the Enterprise Kernel module.

chrisvest committed:
Add PageCache.reportEvents.

With this, client code does not have to depend on, and call out to, the cursor tracer.
So allows entanglement between modules to be reduced.

chrisvest committed:
Add PagedFile.file method, so given a PagedFile you can figure out what file it maps.

ali-ince committed:
Introduce BoltScheduler for simple bolt server thread management

lutovich committed:
Use equivalent of cached thread pool for Bolt workers

Originally `ThreadedWorkerFactory` started thread for every incoming
connection. It has been changed to instead submit tasks to a fixed-size
executor. Such executor is conceptually different from starting new
threads because it can only grow to the upper bound. On-demand threads
had no upper bound.

This commit makes behaviour same as before
(with `new Thread(...).start()`) by using an executor without upper
bound. It should result in less threads being started in total because
some can be reused.

chrisvest committed:
Fix a bug in Format.duration where the empty string would be returned for a value of zero.

Now we instead produce a string that specifies the value zero of the lowest unit.

chrisvest committed:
First draft of the PageCacheWarmer, that profiles and preloads page cache data, in an attempt at warming the page cache up faster after a restart.

ali-ince committed:
Make `BoltScheduler` configurable per bolt connector

ali-ince committed:
Resolve conflicts on rebase

klaren committed:
Do flushing of the page cache in parallel when IOLimiter is unlimited.

digitalstain committed:
Compilation fixes
